blog/2023-10-14-lightweight-cloud-registry-oras/index.mdx

```diff
@@ -0,0 +1,406 @@
+---
+slug: lightweight-cloud-registry-oras
+title: Lightweight Registry with Oras OCI Layouts and Cloud Object Storage
+authors: [oras]
+tags: [oras, community]
+---
+
+## ORAS Lightweight Cloud Registry
+
+Setting up a centralized cloud OCI registry is sometimes too much for small scale tests and proof of concepts!  
+What if I told you there was a lightweight alternative?  
+
+<!--truncate-->
+
+## Everything Should Be a File
+
+One of the great Unix philosophies is the concept of making everything a file.
+Modems, keyboards, printers, disk drives, serial interfaces, configuration, kernel parameteres, IPC, everything and anything is a file that you can `read()` and/or `write()` to.
+
+In a nutshell this separates a large many concerns to the file system layer instead of your program.  
+It allows, among other things, plugging stuff into any program without writing special integration code.  
+That code was already written in the "driver" for that "device", why repeat yourself in the program?
+
+In General: **programs write and read files**  
+Making everything a file means that you can generally interface with every program.
+
+### Oras is [EIAF](https://en.wikipedia.org/wiki/Everything_is_a_file) Compatible
+
+`oras` has a `--oci-layout`
+
+Instead of talking to an upstream service over `http/https`, `oras` can create the OCI artifact structure as files and directories.
+
+Most (if not every) operation that can be performed on an upstream registry can also be performed locally as an OCI directory structure.
+
+### Cloud Object Storage is [EIAF](https://en.wikipedia.org/wiki/Everything_is_a_file) Compatible
+
+The file system is the foundation on which EIAF rests upon.  
+They are special components that convert media (CD, Tape, HDD, Controllers, etc.) into interface layers on which you can read/write files to.  
+
+If you can mount **it** as a file system, you got EIAF.
+
+Your preferred Cloud provider should have its own FUSE driver for their object storage offering, and if it doesn't, there is probably an alternative.
+* [Google Cloud Storage](https://github.com/GoogleCloudPlatform/gcsfuse)
+* [Azure Blob Storage](https://github.com/Azure/azure-storage-fuse)
+* [AWS Simple Storage Service](https://github.com/s3fs-fuse/s3fs-fuse) (Unofficial)
+
+---
+
+It is useful to think of COS as `key: data` databases instead of the file systems they pretend to be.  

```

Comment:
```suggestion
It is useful to think of Cloud Object Storage (COS) as `key: data` databases instead of the file systems they pretend to be.  

```

Reply:
Damn, I was unaware of IBM's [COS](https://www.ibm.com/products/cloud-object-storage), I wanted to make the article generic to most any Cloud Object Storage solutions with minimal confusion.
My options are: 
* Continue with the term COS, as it is the most generic definition of the technology and all confusion is IBM's fault for naming their product something so generic,
* Change it to something like Object Storage, and specify remote/cloud as appropriate